components.json

```diff
@@ -432,14 +435,17 @@
 		"javadoc": {
 			"title": "JavaDoc",
 			"require": ["markup", "java", "javadoclike"],
-			"peerDependencies": [
+			"modify": [
+				"java"
+			],
```

Comment:
Do we need `modify` here since java is already in `require`?

Reply:
`require` assumes that the required language isn't changed. It's a dependency where the required resource is readonly.

The meaning of `require` before this PR was different (see the initial comment of this PR), so the `modify` is necessary.

Reply:
Is that distinction necessary to capture here? For our purposes (generating a list of dependencies in the correct order), we don't need to know whether it modifies the dependency or not. Are there use cases where the distinction is significant?

Reply:
This is distinction is very important when it comes to reloading. We have to reload resources when loading languages not in batch but incrementally. Examples of incremental language loading are [`loadLanguages`](https://github.com/PrismJS/prism/blob/master/components/index.js) and the AutoLoader.

The problem here is that with the old meaning of `require`, every language reachable from the languages being added has to be reloaded. 
For example, if `c-like`, `js`, `css`, and `markup` are loaded and now we want to add `java`, all 4 languages have to be reloaded. (This is the current implementation of `loadLanguages`)
(A full example as to why is kinda lengthy. Should I do one?)

So, to minimize the number of reloaded languages, I introduced readonly dependencies (do they have a real name?).
For example, we know that Java doesn't modify C-like, so we don't have to reload C-like (and all of the other languages because of C-like).

Reply:
No, I get it, thanks.